src/Illuminate/Container/Container.php

```diff
@@ -1160,7 +1160,14 @@ public function afterResolving($abstract, Closure $callback = null)
         if ($abstract instanceof Closure && is_null($callback)) {
             $this->globalAfterResolvingCallbacks[] = $abstract;
         } else {
-            $this->afterResolvingCallbacks[$abstract][] = $callback;
+            // if the type is resolved, we can already run the callback on the object.
+            if($this->resolved($abstract)) {
+                $object = $this->get($abstract);
+                $this->fireCallbackArray($object, [$callback]);    
+            }else {
```

Comment:
If the callback should still be needed for later execution, we should remove this `else`, so it fires the callback once (since it's already resolved) and store the callback for further executions (if resolved again ?). I was not sure how to do it here 👍